src/txmempool.cpp

```diff
@@ -247,6 +247,24 @@ void CTxMemPool::removeConflicts(const CTransaction &tx, std::list<CTransaction>
     }
 }
 
+void CTxMemPool::removeExpired(unsigned int nBlockHeight)
+{
+    // Remove expired txs from the mempool
+    LOCK(cs);
+    list<CTransaction> transactionsToRemove;
```

Comment:
We're using C++11 so the following should work fine:
```
std::map<uint256, CTxMemPoolEntry>::iterator it = mapTx.begin();
while (it != mapTx.end()) {
    if (IsExpired(it->second.GetTx(), nBlockHeight)) {
       it = mapTx.erase(it);
    } else {
       ++it;
    }
}
```

Reply:
No, we need to call the mempool's `remove()` function in order to ensure that all intended side-effects occur. Also, this code will need to change when the PR is rebased on master, where we pulled in an upstream PR that changed `mapTx` to be a `boost::multi_index_container`.

Reply:
Ah I see, changed.